gwpy/timeseries/timeseries.py

```diff
@@ -1388,28 +1388,143 @@ def demodulate(self, f, stride=1, exp=False, deg=True):
         >>> ax = plot.gca()
         >>> ax.set_ylabel('Strain Amplitude at 331.3 Hz')
         >>> plot.show()
+
+        See also
+        --------
+        TimeSeries.heterodyne
+            for the underlying heterodyne detection method
         """
+        # stride through the TimeSeries and heterodyne at a single frequency
+        phase = (2 * numpy.pi * f *
+                 self.dt.decompose().value *
```

Comment:
@mattpitkin, after thinking about it, the following change is needed to prevent a memory leak due to double-precision floats:

```suggestion
                 self.dt.astype('float32').decompose().value *
```